xdsl/interpreters/pdl_interp.py

```diff
@@ -131,6 +133,107 @@ def run_getdefiningop(
         )
         return (args[0].owner,)
 
+    @impl_terminator(pdl_interp.CheckOperationNameOp)
+    def run_checkoperationname(
+        self,
+        interpreter: Interpreter,
+        op: pdl_interp.CheckOperationNameOp,
+        args: tuple[Any, ...],
+    ) -> tuple[Any, ...]:
+        assert len(args) > 0
```

Comment:
Nit: as a code style we try to stick to checking whether a collection is truthy, rather than first calculating the length, which could be O(n), then comparing with 0
```suggestion
        assert args
```

Reply:
and below

Reply:
Right, sorry, you'd told me this before.
In this case the checks should actually check for a specific length, mostly `== 1`. Updated as such.